packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -606,7 +608,31 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
                 uint256(0), // value
                 uint64(SYSTEM_DEPOSIT_GAS_LIMIT), // gasLimit
                 false, // isCreation,
-                abi.encodeCall(IL1Block.setGasPayingToken, (_token, _decimals, _name, _symbol))
+                abi.encodeCall(IL1Block.setConfig, (_type, _value))
+            )
+        );
+    }
+
+    /// @notice Updates a L2 predeploy
+    ///         TODO: can skip system config likely and just
+    ///         if (msg.sender != ISuperchainConfig.upgrader()) revert Unauthorized();
+    ///         this removes the need to have the system config know about the superchain config
+    function upgrade(address payable _proxy, address _implementation) external {
+        if (msg.sender != address(systemConfig)) revert Unauthorized();
```

Comment:
This should be updated to expect a call directly and from the superchain config upgrader

Reply:
https://github.com/ethereum-optimism/optimism/blob/feat/holocene-contracts/packages/contracts-bedrock/src/L1/OptimismPortal2.sol#L637